src/dbnode/storage/index/postings_list_cache_lru.go

```diff
@@ -147,13 +139,13 @@ func (c *postingsListLRU) Add(
 // Get looks up a key's value from the cache.
 func (c *postingsListLRU) Get(
 	segmentUUID uuid.UUID,
-	pattern string,
+	query PostingsListCacheQuery,
 	patternType PatternType,
 ) (postings.List, bool) {
+	newKey := newKey(query, patternType)
```

Comment:
Won't you create this here for no reason if `uuidArray` is not present in `items`? Looks like in `Add` you always have to generate `newKey` so it's ok there, but might be unnecessary both here and in `Remove`

Reply:
yeah but its a stack alloc so I think it doesnt matter much

Reply:
no alloc > stack alloc ;)